backend/tilavarauspalvelu/models/reservation/actions.py

```diff
@@ -290,3 +291,12 @@ def refund_paid_reservation(self) -> None:
 
         payment_order.status = OrderStatus.REFUNDED
         payment_order.save(update_fields=["refund_id", "status"])
+
+    def overlapping_reservations(self) -> ReservationQuerySet:
+        """Find all reservations that overlap with this reservation."""
+        reservation_unit = self.reservation.reservation_units.first()
+        return Reservation.objects.overlapping_reservations(
+            reservation_unit=reservation_unit,
+            begin=self.reservation.begin,
+            end=self.reservation.end,
+        ).exclude(id=self.reservation.id)
```

Comment:
Since this is an action of Reservation, should we pass buffers from the Reservation to `overlapping_reservations`? If no buffers is passed, the ReservationUnit buffers are used. 🤔

Reply:
Fixed